client/components/CreateProduct.js

```diff
@@ -0,0 +1,118 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import { Link } from "react-router-dom";
+import { createProductThunk } from "../store/products";
+
+class CreateProduct extends Component {
+   constructor() {
+      super();
+   }
+
+   handleChange = (evt) => {
+      this.setState({
+         [evt.target.name]: evt.target.value,
+      });
+   };
+
+   handleSubmit = (evt) => {
+      evt.preventDefault();
+      const { products, createProduct } = this.props;
+      createProduct({ ...products, ...this.state });
+
+      //TODO: clear the create form
+      // const form = document.getElementById("create-form");
+      // form.childNodes.forEach((input) => {
+      //    input.value = "";
+      // });
+   };
+
+   render() {
+      const { products } = this.props;
+      const { productName, brand, description, price, inventoryQty } = products;
```

Comment:
I'm not sure `products` is what you think it is here. Creating a product shouldn't need to rely on the redux store for anything